redis/src/test/scala/docs/scaladsl/RedisFlowSpec.scala

```diff
@@ -0,0 +1,200 @@
+import org.specs2.mutable.Specification
+import org.specs2.specification.BeforeAfterAll
+
+import scala.collection.immutable
+import scala.collection.immutable.Seq
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, Future}
+import scala.util.Random
+
+class RedisFlowSpec extends Specification with BeforeAfterAll with RedisSupport {
+
+  sequential
+
+  "scaladsl.RedisFlow" should {
+
+    "insert single key/value" in assertAllStagesStopped {
+
+      val key = Random.alphanumeric.take(10).mkString("")
+
+      val result = Source
+        .single(RedisKeyValue(key, "value1"))
+        .via(RedisFlow.set(1, connection))
+        .runWith(Sink.head[RedisOperationResult[RedisKeyValue[String, String], String]])
+      Await.result(result, Duration(5, TimeUnit.SECONDS)).result.get shouldEqual "OK"
+    }
+
+    "append single value" in assertAllStagesStopped {
+
+      val key = Random.alphanumeric.take(10).mkString("")
+      val result = Source
+        .single(RedisKeyValue(key, "value1"))
+        .via(RedisFlow.append(1, connection))
+        .runWith(Sink.head[RedisOperationResult[RedisKeyValue[String, String], Long]])
+      Await.result(result, Duration(5, TimeUnit.SECONDS)).result.get shouldEqual 6L
+    }
+
+    "insert bulk of key/value" in assertAllStagesStopped {
+
+      val sets = Seq[RedisKeyValue[String, String]](RedisKeyValue("key1", "value1"),
+                                                    RedisKeyValue("key2", "value2"),
+                                                    RedisKeyValue("key3", "value3"))
+
+      val result = Source
+        .fromIterator(() => sets.iterator)
+        .grouped(2)
+        .via(RedisFlow.mset(2, connection))
+        .runWith(Sink.head[RedisOperationResult[Seq[RedisKeyValue[String, String]], String]])
+
+      Await.result(result, Duration(5, TimeUnit.SECONDS)).result.isSuccess shouldEqual true
+    }
+
+    "insert multiple values to single key with lpush" in assertAllStagesStopped {
+
+      val key = Random.alphanumeric.take(10).mkString("")
+
+      val result = Source
+        .single(RedisKeyValues(key, Array("1", "2", "3")))
+        .via(RedisFlow.lpush(1, connection))
+        .runWith(Sink.head[RedisOperationResult[RedisKeyValues[String, String], Long]])
+
+      Await.result(result, Duration(5, TimeUnit.SECONDS)).result.get shouldEqual 3L
+
+    }
+
+    "implement pub/sub for single topic and return notfication of single consumer" in {
+      val topic = "topic20"
+      RedisSource.subscribe(Seq(topic), pubSub).runWith(Sink.ignore)
+
+      val result = Source
+        .single("Bla")
+        .map(f => RedisPubSub(topic, f))
+        .via(RedisFlow.publish[String, String](1, redisClient.connectPubSub().async().getStatefulConnection))
+        .runWith(Sink.head[RedisOperationResult[RedisPubSub[String, String], Long]])
+      redisClient.connectPubSub().sync().unsubscribe(topic)
```

Comment:
This here could be avoided if the stage did unsubscribe on completion.

Reply:
Yes. will create From state for unsubscribe.